test/core/security/grpc_tls_certificate_provider_test.cc

```diff
@@ -495,8 +495,18 @@ TEST_F(GrpcTlsCertificateProviderTest,
   CancelWatch(watcher_state_1);
 }
 
-}  // namespace testing
+TEST_F(GrpcTlsCertificateProviderTest, ValidPrivateKeyCertChainPair) {
+  absl::Status match_status(
+      PrivateKeyPublicKeyMatches(private_key_2_, cert_chain_2_));
+  EXPECT_TRUE(match_status.ok());
+}
 
+TEST_F(GrpcTlsCertificateProviderTest, InvalidPrivateKeyCertChainPair) {
+  absl::Status match_status(
+      PrivateKeyPublicKeyMatches(private_key_2_, cert_chain_));
+  EXPECT_FALSE(match_status.ok());
```

Comment:
Please update test to add check for error message and type
absl::Status status = PrivateKeyPublicKeyMatches(...);
EXPECT_EQ(status.status().code(), absl::StatusCode::kInvalidArgument);
EXPECT_EQ(provider.status().message(), "Invalid credentials pair.");

I haven't looked at the implementation code properly. But I see there are several failure cases, before we do the comparsion. It might be good to include test cases for those too.

Reply:
Noted!